test/DynamoCoreTests/IconsTests.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Dynamo.Models;
+using Dynamo.Search.SearchElements;
+using Dynamo.UI;
+using Dynamo.Wpf.Services;
+using DynamoShapeManager;
+
+using NUnit.Framework;
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Windows.Media;
+
+namespace Dynamo.Tests
+{
+    [TestFixture]
+    internal class IconsTests
+    {
+        private static string geometryFactoryPath = "";
+        private IEnumerable searchEntries;
+        private IconServices iconServices = new IconServices();
+
+        private static string ResourcePath = String.Empty;
+
+        [SetUp]
+        public void InitializeDirectoryPaths()
+        {
+            var assemblyDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            var relativeDir = Path.Combine(assemblyDir, @"..\..\..\src\Resources");
+            var absoluteDir = Path.GetFullPath(relativeDir);
+            Assert.IsFalse(string.IsNullOrEmpty(absoluteDir));
+            Assert.IsTrue(Directory.Exists(absoluteDir));
+            ResourcePath = absoluteDir;
+        }
+
+        [SetUp]
+        public void PreloadShapeManager()
```

Comment:
Please merge both `InitializeDirectoryPaths` and `PreloadShapeManager` into one method called `SetupIconTests` and have another one called `TearDownIconTests` (which is going to be empty I guess, or it can mark `DynamoModel` or `Preloader` instances as `null`).